Profile page restyling

ticruz38 opened the pull request:
Related to #390 and #393

Following and followers count are now part of the tabs with a dedicated feed under the profile page information.

#393 is not totally complete, the card still has the "about" for each followers/following profile, I thought it was nice to have it and just waited for your confirmation to actually remove it.

#394 with the relay feed redesign is not part of this PR yet.


<img width="1012" alt="Screenshot 2024-10-16 at 16 39 56" src="https://github.com/user-attachments/assets/3f8b9c55-ef18-4776-a6b6-7beac3e6b242">
<img width="977" alt="Screenshot 2024-10-16 at 16 39 50" src="https://github.com/user-attachments/assets/32b37c37-31bc-4e77-8ea7-dcfaa6ad6ffa">
<img width="1093" alt="Screenshot 2024-10-16 at 16 39 41" src="https://github.com/user-attachments/assets/be5defee-10e0-498c-b5c9-3f9abababcfc">
<img width="1086" alt="Screenshot 2024-10-16 at 16 39 33" src="https://github.com/user-attachments/assets/ae1adf92-4bc5-4e88-927a-aa960b24a015">

staab commented on src/app/shared/PersonSummary.svelte:
```diff
@@ -28,9 +36,17 @@
-        </slot>
+        <div class="flex items-center justify-end gap-2">
+          {#if $following}
+            <Anchor button low class="border-none bg-tinted-800-d" on:click={e => unfollow(pubkey)}
```
This should use `on:click={unfollowPerson}`

staab commented on src/app/shared/PersonSummary.svelte:
```diff
@@ -28,9 +36,17 @@
+            <Anchor button low class="border-none bg-tinted-800-d" on:click={e => unfollow(pubkey)}
+              >Followed</Anchor>
+          {:else}
+            <Anchor button accent on:click={e => follow(tagPubkey(pubkey))}>Follow</Anchor>
```
This should use `on:click={doFollow}`

staab commented on src/app/views/PersonDetail.svelte:
```diff
@@ -57,11 +84,32 @@
     .qp({splits: [tagZapSplit(pubkey)]})
     .toString()
 
+  $: tabs = [
+    "notes",
+    "likes",
+    "collections",
+    "relays",
+    "following -" + $followsCount,
+    "followers " + $followersCount + "+",
+  ].filter(identity)
```
Instead of interpolating the strings, use the `tab` slot in the `Tabs` component to customize the display. See Notifications.svelte for an example. You could also use the same badge style.

staab commented on src/app/views/PersonDetail.svelte:
```diff
@@ -73,44 +121,63 @@
+    <div>
+      <PersonCircle {pubkey} class="mt-1 h-12 w-12 sm:h-32 sm:w-32" />
+      {#each actions as action (action.label)}
+        <div animate:flip={{delay: 200, duration: 1000, easing: bounceInOut}}>
```
This animation is really funky. Maybe use `transition:slide` instead. Also, there's no need to iterate over actions here, just use conditionals, the extra complexity isn't worth it.

https://github.com/user-attachments/assets/6f6a78f9-91be-4689-85e6-e8717386ddd9

ticruz38 replied:
yes the code is clunky, the only reason to use a keyed each here is to have button message gently take the empty space from the disappearing follow button by sliding in.
Not sure I'll get this same behaviour with transition:slide

ticruz38 replied:
it's true that the animation does not look that good anyway

staab reviewed (changes requested):
Thanks! Some things I noticed:

- [ ] NIP05's have opacity applied, I think they'd look better without opacity. Either daniele's icon or an icon to the left of the nip05 would also be nice.
- [ ] Line 167 of PersonDetail has an empty div that's messing up spacing
- [ ] There should be a bullet or something between nip05 and npub on card headers
- [ ] Now that we're not showing the wot number, let's put the wot arc in line with the username, since it's getting pushed way over to the right by the npub
- [ ] Let's keep follow/unfollow next to "message", daniele's objections notwithstanding
- [ ] Message/follow needs to be replaced by "edit" on the user's own profile page
- [ ] Font size on the user name could be a little bigger
- [ ] Profiles often get opened in modals, maybe use AltColor to make the top part still look like a card
- [ ] Follow/follower counts should be faded and separated by a bullet as in the mockup.
- [ ] The "message" action in the overflow menu can be removed since it's on the main card area already.
- [ ] The wot arc seems to be missing on the profile detail page

staab commented:
Also, I like keeping the about on the profile cards, good choice.

ticruz38 commented:
- [x] Removed opacity and added the fa-check icons (could not find Daniele's icon)
- [x] Line 167 of PersonDetail is removed
- [x] Added the bullet character with opacity-75 between nip05 and npub on card headers
- [x] Wot arch is inline with user name
- [x] follow/unfollow button next to "message"
- [x] Message/follow replaced by "edit" on the user's own profile page, Edit action removed from the dropdown as well
- [x] Font size is now text-xl
- [x] ProfileDetails is now wrapped in the Altcolor.svelte utility component, like notes
- [x] Follow/follower counts are now displayed in a Badge, like notifications tabs.
- [x] The "message" action removed from the overflow menu
- [x] Wot arc back next to the user name, scaled up to match the name font-size

staab commented:
Looks good, I made a few tweaks of my own and pushed to dev. Just a couple final comments.

ticruz38 commented on src/app/shared/PersonActions.svelte:
```diff
@@ -22,26 +18,12 @@
-        ),
-      )
-      .open()
-
```
it should be aligned now

staab commented:
Cool, mind rebasing so I can review?

staab commented on src/app/shared/PersonName.svelte:
```diff
@@ -60,11 +60,10 @@
-          <span class="opacity-50 hidden xs:block">-</span>
-        </PersonHandle>
-        <p class="whitespace-nowrap opacity-50">{npubDisplay}</p>
+      <div class="flex flex-row items-center gap-2">
+        <PersonHandle class="whitespace-nowrap text-xs text-accent" {pubkey} />
+        <span class="text-xs opacity-75">•</span>
+        <small class="whitespace-nowrap">{npubDisplay}</small>
```
I made this change because if there's no handle the bullet is just floating out in space, you'll want to adopt my version here.

ticruz38 replied:
messed up the merge, probably a good argument in favor of rebase here

staab commented on src/app/views/PersonDetail.svelte:
```diff
@@ -197,6 +197,18 @@
       <div class="-ml-16 flex flex-grow flex-col gap-4 xs:ml-0">
         <PersonAbout class="font-thin opacity-75" {pubkey} />
       </div>
+      <div class="-ml-16 flex flex-grow flex-col gap-4 xs:ml-0">
+        <PersonAbout class="font-thin opacity-75" {pubkey} />
+      </div>
+      {#if $profile?.website}
+        <Anchor
+          external
+          class="flex items-center gap-2 overflow-hidden overflow-ellipsis whitespace-nowrap"
+          href={ensureProto($profile.website)}>
+          <i class="fa fa-link text-accent" />
+          {stripProtocol($profile.website)}
+        </Anchor>
+      {/if}
```
This isn't right, I moved website elsewhere, and personabout is now duplicated.

ticruz38 replied:
I went through your commits, if I'm not mistaking, website is now inline at the tail end of PersonHandle and zapDisplay.

It will go hidden if PersonHandle and zapDisplay are too large, it is hidden on my profile for example.

<img width="641" alt="Screenshot 2024-10-25 at 11 15 02" src="https://github.com/user-attachments/assets/ce3e4ee2-5b88-49b8-91bf-58d763313dea">

The row is 80% of the container width atm, I can make it full size and add the flex-wrap condition on the container.
It will push the website to the next line.

<img width="637" alt="Screenshot 2024-10-25 at 11 22 54" src="https://github.com/user-attachments/assets/2421ba67-16a3-4fa2-bdee-8afef394b162">

Let me know your preference and I'll push the commit eventually

staab reviewed (changes requested):
> Merge remote-tracking branch 'upstream/dev' into feat/user-profile

Merge is fine, but rebase is preferred, for future reference.
